### PR TITLE
Add toLowerCase helper to StringUtil for working with CaseMapping

### PR DESCRIPTION
**Pull request by kashike**

### CODE REVIEW

File: src/main/java/org/kitteh/irc/client/library/util/StringUtil.java [hunk 1]
```diff
@@ -120,4 +124,37 @@ public static String makeRainbow(@Nonnull String message, @Nonnull Format[] colo
...
     }
+
+    /**
+     * Converts a given String to lowercase per spec.
+     *
+     * @param cliented the client provider
+     * @param input string to be lowercased
+     * @return lowercased string
+     * @throws IllegalArgumentException if client provider is null
+     * @throws IllegalArgumentException if input is null
```
Comment (mbax) [inline src/main/java/org/kitteh/irc/client/library/util/StringUtil.java]: Maybe "if either parameter is null" ?

Also, thoughts on nullchecking the result of `#getClient()`? KICL doesn't do it but other things might

Comment (kashike) [inline src/main/java/org/kitteh/irc/client/library/util/StringUtil.java]: The result of `getClient()` is already null-checked.

Comment (mbax) [inline src/main/java/org/kitteh/irc/client/library/util/StringUtil.java]: ...oh

**Comment by mbax** [comment]
Thanks!

**Pull request merged**